libmpv/csharp/Form1.cs

```diff
@@ -0,0 +1,167 @@
+        private delegate int MpvSetProperty(IntPtr mpvHandle, byte[] name, int format, ref byte[] data);
+        private MpvSetProperty _mpvSetProperty;
+
+        public Form1()
+        {
+            InitializeComponent();
+        }
+
+        private object GetDllType(Type type, string name)
+        {
+            IntPtr address = GetProcAddress(_libMpvDll, name);
+            if (address != IntPtr.Zero)
+                return Marshal.GetDelegateForFunctionPointer(address, type);
+            return null;
+        }
+
+        private void LoadMpvDynamic()
+        {
+            _libMpvDll = LoadLibrary("mpv-1.dll"); // The dll is included in the DEV builds by lachs0r: https://mpv.srsfckn.biz/
+            _mpvCreate = (MpvCreate)GetDllType(typeof(MpvCreate), "mpv_create");
+            _mpvInitialize = (MpvInitialize)GetDllType(typeof(MpvInitialize), "mpv_initialize");
+            _mpvTerminateDestroy = (MpvTerminateDestroy)GetDllType(typeof(MpvTerminateDestroy), "mpv_terminate_destroy");
+            _mpvCommand = (MpvCommand)GetDllType(typeof(MpvCommand), "mpv_command");
+            _mpvSetOption = (MpvSetOption)GetDllType(typeof(MpvSetOption), "mpv_set_option");
+            _mpvSetOptionString = (MpvSetOptionString)GetDllType(typeof(MpvSetOptionString), "mpv_set_option_string");
+            _mpvGetPropertyString = (MpvGetPropertystring)GetDllType(typeof(MpvGetPropertystring), "mpv_get_property");
+            _mpvSetProperty = (MpvSetProperty)GetDllType(typeof(MpvSetProperty), "mpv_set_property");
+        }
+
+        public void Pause()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            var bytes = Encoding.UTF8.GetBytes("yes\0");
+            _mpvSetProperty(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref bytes);
+        }
+
+        private void Play()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            var bytes = Encoding.UTF8.GetBytes("no\0");
+            _mpvSetProperty(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref bytes);
+        }
+
+        public bool IsPaused()
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return true;
+
+            IntPtr lpBuffer = Marshal.AllocHGlobal(10);
+            _mpvGetPropertyString(_mpvHandle, Encoding.UTF8.GetBytes("pause\0"), MpvFormatString, ref lpBuffer);
+            return Marshal.PtrToStringAnsi(lpBuffer) == "yes";
+        }
+
+        public void SetTime(double value)
+        {
+            if (_mpvHandle == IntPtr.Zero)
+                return;
+
+            _mpvCommand(_mpvHandle, new[] { "seek", value.ToString(CultureInfo.InvariantCulture), "absolute", null });
+        }
+
+        private void buttonPlay_Click(object sender, EventArgs e)
+        {
+            if (_mpvHandle != IntPtr.Zero)
+                _mpvTerminateDestroy(_mpvHandle);
+
+            LoadMpvDynamic();
```

Comment:
It's not necessary to destroy and recreate mpv every time a new file has to be loaded. The ``loadfile`` command can be sent at any time and mpv will happily idle in the background when there are no files to play. Unless there's a specific reason to destroy and recreate mpv (such as changing options that can't be set on the fly,) the mpv handle should probably just be created once when the form is shown.